### ISSUE TITLE
loss doesn't converge on one scale

### ISSUE THREAD

**Issue by xi3299** [eric612/MobileNet-YOLO#141]
Model : tiny-yolov3

Backbone : darknet tiny

Questions :
Hello,
I want to train tiny-yolov3 for person class of COCO. The batch_size has been set to 4*16=64, after have trained 40,000iterations, the loss does not decrease any more. Especially in the scale of 26*26, for the scale of 13*13 the loss looks like decreased normally. 

> 
> I0726 17:08:20.711820 23984 solver.cpp:254] Iteration 41610 (0.376511 iter/s, 26.5597s/10 iters), loss = 25.2371
> I0726 17:08:20.711820 23984 solver.cpp:273]     Train net output #0: det_loss1 = 1.02525 (* 1 = 1.02525 loss)
> I0726 17:08:20.711820 23984 solver.cpp:273]     Train net output #1: det_loss2 = 22.241 (* 1 = 22.241 loss)
> I0726 17:08:20.711820 23984 sgd_solver.cpp:121] Iteration 41610, lr = 1e-05

I am now wondering how to set correctly the parameters in solver.prototxt file. Is this due to the high learning rate?
Here, I put the solver.prototxt and deploy.prototxt.
@eric612 
[solver2.txt](https://github.com/eric612/MobileNet-YOLO/files/3436318/solver2.txt)
[tiny-yolov3-deploy.txt](https://github.com/eric612/MobileNet-YOLO/files/3436319/tiny-yolov3-deploy.txt)
Thanks.

**Comment by xi3299** [comment]
@eric612 Sorry to cue you again.
Would you some ideas with this issue? I am wondering is there a problem with the selection of anchors?
Please.

**Comment by eric612** [comment]
@xi3299 ,
I can't not find any crystal problem of the prototxt which you provide , but , one little question is it looks like there is no small object in your dataset ?

It is recommend to upload training log .

**Comment by xi3299** [comment]
Thank you for your response.
[training_log.txt](https://github.com/eric612/MobileNet-YOLO/files/3473137/training_log.txt)
Here, is the part of the training log. 
[slover.txt](https://github.com/eric612/MobileNet-YOLO/files/3473145/slover.txt)
And I've also changed the parametr of stagelr and stageiter for the strategy of learning rate. 

Thank you for your time!

**Comment by xi3299** [comment]
@eric612 Thank you for your response.
[training_log.txt](https://github.com/eric612/MobileNet-YOLO/files/3473155/training_log.txt)
Here, is the part of the training log.
[slover.txt](https://github.com/eric612/MobileNet-YOLO/files/3473157/slover.txt)
And I've also changed the parametr of stagelr and stageiter for the strategy of learning rate.

Thank you for your time!
> 
> 
> @xi3299 ,
> I can't not find any crystal problem of the prototxt which you provide , but , it looks like there is no small object in your dataset ?
> 
> It is recommend to upload training log .

**Comment by eric612** [comment]
The count distribution was unbalance between two scales , here has two ways to improve this problem: 
* if you have set expand_param , decrease max_expand_ratio or delete expand_param , and increase yolo_data_jitter
* recalculation bounding box biases

**Comment by xi3299** [comment]
> 
> 
> The count distribution was unbalance between two scales , here has two ways to improve this problem:
> 
>     * if you have set expand_param , decrease max_expand_ratio or delete expand_param , and increase yolo_data_jitter
> 
>     * recalculation bounding box biases

@eric612 Thank you so much. 
-For this solution, I didn't set expand_param and I increased yolo_data_jitter from 0.3 to 0.5. But can I ask why it should be increased?
-The bounding box were calculated by this way. #92  Although the anchors do not look so appropriate, I didn't find the other better way to recalculate them.

**Comment by eric612** [comment]
@xi3299 
Or you can try to recalculate anchors on resolution 320x320 or lower resolution , and also use this resolution with training

**Comment by xi3299** [comment]
> 
> 
> @xi3299
> Or you can try to recalculate anchors on resolution 320x320 or lower resolution , and also use this resolution with training

Alright, I will try this way. Thanks again. ^^

**Comment by eric612** [comment]
One more thing , if your dataset only have one class , the avg_cat would approximate to 1.0

**Comment by xi3299** [comment]
> 
> 
> One more thing , if your dataset only have one class , the avg_cat would approximate to 1.0

- Yes, there is only one class in my dataset. But the avg_cat is about 0.4 and it's not really stable. So is there something wrong? 
- And when I use my model to detect on an image, the label is always "aeroplane", but I am pretty sure  the labelmap.prototxt is modified in a right way. Like the following lines. 

> item {
  name: "none_of_the_above"
  label: 0
  display_name: "background"
}
item {
  name: "1"
  label: 1
  display_name: "person"
}
- By the way, I changed the resolution to 320*320 for training and increased yolo_data_jitter , the accuracy now is improved. So really thank you again for your suggestion.
